ompi/mpi/fortran/use-mpi-f08/pmpi-f-interfaces-bind.h

```diff
@@ -98,9 +98,9 @@ end subroutine pompi_buffer_attach_f
 ! Note, too, we don't need a "p" version of the C implementation
 ! function -- Fortran's interfaces provide MPI_ and PMPI_ names for
 ! us; they can just both be bound to the same back-end
-! ompi_buffer_detach_f08 C function.
+! pompi_buffer_detach_f08 C function.
 subroutine pompi_buffer_detach_f(buffer_addr,size,ierror) &
-   BIND(C, name="ompi_buffer_detach_f08")
+   BIND(C, name="pompi_buffer_detach_f08")
```

Comment:
I don't know whether there is a real "pompi_buffer_detach_f08' function at Open MPI. This is at Fortran 2008 support model. It is rarely used and skipped when building by gfortran. 
The fix is just about bind name which is different from function name before. At ifort, it passed. But there is more strict syntax at xlf.
I am not familiar with GitHub and will rebase this PR later. Thanks very much. @jsquyres 
Error message from xlf:
"pmpi-f-interfaces-bind.h", line 100.1: 1516-194 (S) A binding label must not be the same as the name of a program unit, common block, external procedure, and another binding label, ignoring differences in case.